src/expression.js

```diff
@@ -234,19 +238,30 @@ pp.parseSubscripts = function(base, startPos, startLoc, noCalls) {
     } else if (this.eat(tt.bracketL)) {
       let node = this.startNodeAt(startPos, startLoc)
       node.object = base
-      node.property = this.parseExpression()
+      node.property = this.parseExpression(false, refContextDestructuringErrors)
       node.computed = true
       this.expect(tt.bracketR)
       base = this.finishNode(node, "MemberExpression")
     } else if (!noCalls && this.eat(tt.parenL)) {
+      let refDestructuringErrors = new DestructuringErrors
+      let exprList = this.parseExprList(tt.parenR, false, false, refDestructuringErrors)
+      if (maybeAsyncArrow && !this.canInsertSemicolon() && this.eat(tt.arrow)) {
```

Comment:
Argh, never mind, the original expression is inspecting the _node_, not the token state. Ignore me.